### PR TITLE
Add hammer plugins template tests

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - string:
+        name: hammer_plugin_branch
+        default: master
+        description: 'git branch of the hammer plugin to base on'
+
+    dsl:
+      !include-raw:
+    - pipelines/lib/rvm.groovy
+    - pipelines/lib/git.groovy
+    - pipelines/test/hammer_plugin_template_test.groovy
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml]: Please indent this 2 more steps.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/hammer_plugin_template_test.groovy [hunk 2]
```diff
@@ -0,0 +1,48 @@
+def version_map = branch_map[env.gitlabTargetBranch]
+def ruby = version_map['ruby']
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/hammer_plugin_template_test.groovy]: This version map does not exist for our CI system.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/hammer_plugin_template_test.groovy [hunk 3]
```diff
@@ -0,0 +1,48 @@
+def version_map = branch_map[env.gitlabTargetBranch]
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/hammer_plugin_template_test.groovy]: We don't use gitlab.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/hammer_plugin_template_test.groovy [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+def ruby = version_map['ruby']
+
+pipeline {
+
+  agent { label 'sat6-build' }
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/hammer_plugin_template_test.groovy]: This agent does not exist. You can drop this line.

**Comment by shiramax** [comment]
@ehelms thanks for the review! done :)

**Comment by ehelms** [comment]
@shiramax Are you trying to setup PR testing for this project?

**Comment by shiramax** [comment]
@ehelms I want to create a template for all hammer plugins.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
+- job-template:
+    name: test_hammer_plugin_{name}_{branch}
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml]: To follow our conventions this should become:

{name}-pr-test

Example: https://ci.theforeman.org/job/katello-pr-test/

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    project-type: pipeline
+    sandbox: true
+    parameters:
+    - string:
+        name: hammer_plugin_url
+        description: 'git url of the hammer plugin repo'
+    - string:
+        name: hammer_plugin_branch
+        default: master
+        description: 'git branch of the hammer plugin to base on'
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml]: These parameters are not needed.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    dsl:
+      !include-raw:
+      - pipelines/lib/rvm.groovy
+      - pipelines/lib/git.groovy
+      - pipelines/test/hammer_plugin_template_test.groovy
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml]: I think you could drop `template` in the name here.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+- project:
+    name: hammer_cli_foreman_kubevirt
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml]: Let's use dashes please, as that's our typical convention and the repository name

**Comment by ehelms** [comment]
Here would be my example to start with, this captures both Rubies available across EL and Debian. This is untested, but once you make updates we can collaborate on testing it in the environment.

```
pipeline {
    agent none

    options {
        timeout(time: 1, unit: 'HOURS')
    }

    stages {
        stage('Test') {
            matrix {
                agent any
                axes {
                    axis {
                        name 'ruby'
                        values '2.5', '2.6'
                    }
                }
                stages {
                    stage('Setup Git Repos') {
                        steps {
                            ghprb_git_checkout()
                            sh "cp Gemfile Gemfile.${ruby}"
                            sh "\\\"gem 'hammer_cli_foreman', :git => 'https://github.com/theforeman/hammer-cli-foreman.git'\\\" > Gemfile.local"
                            sh "\\\"gem 'hammer_cli', :git => 'https://github.com/theforeman/hammer-cli.git\\\'" >> Gemfile.local"
                        }
                    }
                    stage("Setup RVM") {
                        steps {
                            configureRVM(ruby, ruby)
                        }
                    }
                    stage('Install dependencies') {
                        steps {
                            withRVM(["bundle install --gemfile Gemfile.${ruby}"], ruby, "${ruby}")
                        }
                    }
                    stage('Run Tests') {
                        steps {
                            withRVM(["bundle exec rake ci:setup:minitest test TESTOPTS='-v' --trace Gemfile.${ruby}"], ruby, ruby)
                        }
                    }
                }
                post {
                    always {
                        archiveArtifacts artifacts: "Gemfile*lock"
                        junit keepLongStdio: true, testResults: 'jenkins/reports/unit/*.xml'
                        cleanupRVM(ruby, "${ruby}-${PUPPET_VERSION}")
                        deleteDir()
                    }
                }
            }
        }
    }
}
```

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml (hunk 6 above)
Comment (shiramax) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml:13]: @ehelms, 
how can I know that the hammer git plugin will be installed?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml (hunk 8 above)
Comment (shiramax) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_hammer_plugin.yaml:22]: @ehelms , are you sure? the gem is with an underscore,

**Comment by shiramax** [comment]
closing this one for now.

**Pull request closed without merging**